ui/src/pageLayout/utils/index.test.ts

```diff
@@ -0,0 +1,90 @@
+// Constants
+import {HOMEPAGE_PATHNAME} from 'src/shared/constants'
+
+// Utils
+import {getNavItemActivation} from 'src/pageLayout/utils'
+
+describe('pageLayout.utils.getNavItemActivation', () => {
+  it('matches the homepage when the location pathname has fewer than 3 parts', () => {
+    let pathname = ''
+    expect(
+      getNavItemActivation([HOMEPAGE_PATHNAME, 'account'], pathname)
+    ).toEqual(true)
+
+    pathname = '/orgs'
+    expect(
+      getNavItemActivation([HOMEPAGE_PATHNAME, 'account'], pathname)
+    ).toEqual(true)
+
+    pathname = '/orgs/3491cbaef55b4559'
+    expect(
+      getNavItemActivation([HOMEPAGE_PATHNAME, 'account'], pathname)
+    ).toEqual(true)
+
+    pathname = '/orgs/3491cbaef55b4559/some-path'
+    expect(
+      getNavItemActivation([HOMEPAGE_PATHNAME, 'account'], pathname)
+    ).toEqual(false)
+
+    pathname = '/orgs/3491cbaef55b4559'
+    expect(getNavItemActivation(['not-homepage', 'account'], pathname)).toEqual(
+      false
+    )
+
+    pathname = '/new/meme/reviews'
+    expect(
+      getNavItemActivation([HOMEPAGE_PATHNAME, 'account'], pathname)
+    ).toEqual(false)
+  })
+
+  it('matches the given name strictly when it appears after the third slash in the pathname', () => {
+    let given = 'data-explorer'
+    const base = '/orgs/3491cbaef55b4559'
+
+    expect(getNavItemActivation([given], '/orgs/1/data-explorer')).toEqual(true)
+    expect(getNavItemActivation([given], '/new/meme/reviews')).toEqual(false)
+
+    given = 'dashboards'
+    expect(
+      getNavItemActivation([given, 'another-name'], `${base}/${given}`)
+    ).toEqual(true)
+
+    given = 'tasks'
+    expect(
+      getNavItemActivation([given], `${base}/${given}/longer-path`)
+    ).toEqual(true)
+
+    given = 'alerting'
+    expect(
+      getNavItemActivation([given], `/different/basepath/${base}/${given}`)
+    ).toEqual(true)
+
+    given = 'alert-history'
+    expect(getNavItemActivation([given], `/${given}`)).toEqual(false)
+
+    given = 'load-data'
+    expect(getNavItemActivation([given], `/two-slashes/${given}`)).toEqual(
+      false
+    )
+
+    given = 'buckets'
+    expect(getNavItemActivation([given], `${base}${given}/${given}`)).toEqual(
+      true
+    )
+
+    given = 'telegrafs'
+    expect(getNavItemActivation([given], `${base}/${given}${given}`)).toEqual(
+      false
+    )
+
+    given = 'scrapers'
+    expect(
+      getNavItemActivation([given], `${base}/${given}${given}${given}`)
+    ).toEqual(false)
+
+    given = 'tokens'
+    expect(getNavItemActivation([given], `${base}////////${given}`)).toEqual(
+      true
+    )
+  })
+})
```

Comment:
:heart: love these tests!

only nit / style suggestion i could provide is: instead of using variables, just hard code the values in the test - it's a little more repetitive, but i find it makes the tests more explicit and it usually makes the intent of the test and the code a little easier to follow. again, a nit / suggestion, feel free to ignore.

```js
expect(getNavItemActivation(['tokens'], `/orgs/3491cbaef55b4559////////tokens')).toEqual(true)
```